services/web/src/utils/api/request.js

```diff
@@ -0,0 +1,52 @@
+import { API_URL } from 'utils/env';
+import appSession from 'stores/AppSession';
+
+import { ApiError, ApiParseError } from './errors';
+
+export default async function request(options) {
+  const { method = 'GET', path, params } = options;
+  let { body } = options;
+
+  const token = options.token || appSession.token;
+
+  const headers = Object.assign(
+    {
+      Accept: 'application/json',
+      ...(token && {
+        Authorization: `Bearer ${token}`,
+      }),
+    },
+    options.headers
+  );
+
+  const url = new URL(path, API_URL);
+  url.search = new URLSearchParams(params);
+
+  if (body instanceof File) {
+    const data = new FormData();
+    data.append('file', body);
+    body = data;
+  } else if (!(body instanceof FormData)) {
+    body = JSON.stringify(body);
+    headers['Content-Type'] = 'application/json';
+  }
+
+  const res = await fetch(url, {
```

Comment:
Did you test whether file uploading and editing of file upload forms still work?

Reply:
Yep, test it out, works fine. This could even handle more complex stuff in the future.